ci_diff_helper/kokoro.py

```diff
@@ -0,0 +1,353 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Set of utilities for dealing with Kokoro (Google's internal Jenkins).
+
+This module provides a custom configuration type
+:class:`Kokoro` for the `Kokoro`_ CI system.
+
+.. _Kokoro: https://www.cloudbees.com/sites/default/files/2016-jenkins-world-jenkins_inside_google.pdf
+
+This module uses a selection of environment variables to detect the state of
+Kokoro configuration. See :mod:`~ci_diff_helper.environment_vars` for more
+details.
+
+:class:`Kokoro` Configuration Type
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When running in Kokoro, you can automatically detect your
+current environment and get the configuration object:
+
+.. testsetup:: auto-detect
+
+  import os
+  os.environ = {
```

Comment:
1. Shouldn't you save the old `os.environ` and restore it in `testcleanup`?
1. Super-pedantic token @dhermes comment: I usually indent 3 spaces in Sphinx blocks